### ISSUE TITLE
Awake from a parent class is not getting called unless overridden for scene placed network objects

### ISSUE THREAD

**Issue by evbishop** [FirstGearGames/FishNet#554]
**General**
Unity version: 2021.3.18f1
Fish-Networking version: FishNetworking.3.11.10R.Pro
Discord link: https://discord.com/channels/424284635074134018/1177135752107266098

**Description**
Awake from a parent class is not getting called unless overridden for scene placed network objects after the scene containing it is loaded.
I updated from FishNetworking.3.11.5R.Pro and the issue started.
I'm pretty sure this was broken in 3.11.8R as the release notes state: "Fixed rare cases of Awake not calling on parent with multiple inheritance levels.

Parent class example:
```
public abstract class NetworkSingleton<T> : NetworkBehaviour where T : NetworkSingleton<T>
{
    public static T s_instance;

    protected virtual void Awake()
    {
        if (s_instance != null && Instance != this)
            Destroy(gameObject);
        else
            s_instance = this as T;
    }
}
```

Inheriting class example, Awake() will erroneously not get called:
```
public class GlobalNetworkObjectsSpawner : NetworkSingleton<GlobalNetworkObjectsSpawner>
{

}
```

Inheriting class example, Awake() will get called:
```
public class GlobalNetworkObjectsSpawner : NetworkSingleton<GlobalNetworkObjectsSpawner>
{
    protected override void Awake()
    {
        base.Awake();
    }
}
```

**Replication**
1. Make a base abstract NetworkBehaviour class with functionality in Awake()
2. Make a class that inherits from that base class
3. Attach it to a scene placed game object, along with a NetworkObject and NetworkObserver (with a scene condition) components
4. Start the project/load the scene with the game object you made
5. Awake on that object was not called.

**Expected behavior**
Awake() is called on the scene placed network behaviour object that inherits but does not override Awake(), as it was prior to 3.11.8R release.

**Screenshots**
GlobalNetworkObjectsSpawner, placed on an object in the scene:
![image](https://github.com/FirstGearGames/FishNet/assets/16606035/754ffa77-fd8a-42f6-a257-a688a0856c97)

**Comment by evbishop** [comment]
Update: it's not just scene placed network objects, same thing is going on with objects spawned at runtime too

**Comment by FirstGearGames** [comment]
I believe this is something to do with the generic. Thanks for the post.

**Comment by FirstGearGames** [comment]
in `NetworkBehaviourProcessor.cs` change `MethodReference baseAwakeMr = td.GetMethodInBase(base.Session, NetworkBehaviourHelper.AWAKE_METHOD_NAME);`
to `MethodReference baseAwakeMr = td.GetMethodReferenceInBase(base.Session, NetworkBehaviourHelper.AWAKE_METHOD_NAME);`

**Comment by FirstGearGames** [comment]
Resolved in 3.11.11 (official release) and 4.0.4
